src/ds/d400/d400-private.cpp

```diff
@@ -290,6 +290,76 @@ namespace librealsense
             return calc_intrinsic;
         }
 
+        //D405 needs special calculation because the ISP crops the full sensor image using non linear transformation.
+        rs2_intrinsics get_d405_color_stream_intrinsic(const std::vector<uint8_t>& raw_data, uint32_t width, uint32_t height)
+        {
+            // Convert normalized focal lenght and principal point to pixel units (K matrix format)
+            auto k_to_pixels = []( float3x3 & k, ds_rect_resolutions res )
+            {
+                k( 0, 0 ) = k( 0, 0 ) * resolutions_list[res].x / 2.f;  // fx
+                k( 1, 1 ) = k( 1, 1 ) * resolutions_list[res].y / 2.f;  // fy
+                k( 2, 0 ) = ( k( 2, 0 ) + 1 ) * resolutions_list[res].x / 2.f;  // ppx
+                k( 2, 1 ) = ( k( 2, 1 ) + 1 ) * resolutions_list[res].y / 2.f;  // ppy
+            };
+
+            // Scale focal lenght and principal point in pixel units from one resolution to another (K matrix format)
+            auto scale_pixel_k = []( float3x3 & k, ds_rect_resolutions in_res, ds_rect_resolutions out_res )
+            {
+                float scale_x = resolutions_list[out_res].x / static_cast< float >( resolutions_list[in_res].x );
```

Comment:
Need divide by 0 protection?
Could be static analysis error?

Reply:
I had protection at first and removed it as I changed the implementation to use ds_rect_resolutions, but now I see that `width_height_to_ds_rect_resolutions` can fail so I will add it back.

Reply:
Done.
Checking that the resolutions are OK. All `resolutions_list[in_res]` values are hard coded not zero.